app/config/champaign_config.rb

```diff
@@ -0,0 +1,31 @@
+require 'yaml'
+
+module ChampaignConfig
+  extend self
+
+  DEFAULT_YAML = 'config/champaign.yml'
+
+  @location = DEFAULT_YAML
+
+  def oauth_domain_whitelist
+    root['google_oauth_whitelist']['domains']
+  end
+
+  def yaml_location=(path)
+    @location = path
+  end
+
+  def reset!
+    @location = DEFAULT_YAML
+  end
+
+  private
+
+  def data
+    @data = YAML.load_file(@location)
```

Comment:
do you mean to cache the loaded yaml? it doesnt seem like you ever read out of `@data`.